presto-raptorx/src/main/java/com/facebook/presto/raptorx/storage/organization/OrganizationJob.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptorx.storage.organization;
+
+import com.facebook.presto.raptorx.RaptorConnector;
+import com.facebook.presto.raptorx.metadata.ColumnInfo;
+import com.facebook.presto.raptorx.metadata.TableInfo;
+import com.facebook.presto.raptorx.storage.ChunkInfo;
+import com.facebook.presto.raptorx.storage.CompressionType;
+import com.facebook.presto.raptorx.transaction.Transaction;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.transaction.IsolationLevel;
+import com.google.common.collect.ImmutableList;
+import io.airlift.log.Logger;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.facebook.presto.spi.block.SortOrder.ASC_NULLS_FIRST;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Collections.nCopies;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+class OrganizationJob
+        implements Runnable
+{
+    private static final Logger log = Logger.get(OrganizationJob.class);
+
+    private final ChunkCompactor compactor;
+    private final OrganizationSet organizationSet;
+    private final RaptorConnector raptorConnector;
+
+    public OrganizationJob(OrganizationSet organizationSet, ChunkCompactor compactor, RaptorConnector raptorConnector)
+    {
+        this.compactor = requireNonNull(compactor, "compactor is null");
+        this.organizationSet = requireNonNull(organizationSet, "organizationSet is null");
+        this.raptorConnector = requireNonNull(raptorConnector, "transactionManager is null");
+    }
+
+    @Override
+    public void run()
+    {
+        try {
+            runJob(organizationSet.getTableId(), organizationSet.getBucketNumber(), organizationSet.getChunkIds());
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void runJob(long tableId, int bucketNumber, Set<Long> chunkIds)
+            throws IOException
+    {
+        ConnectorTransactionHandle transactionHandle = raptorConnector.beginTransaction(IsolationLevel.REPEATABLE_READ, false);
```

Comment:
We cannot call `raptorConnector.beginTransaction()` on worker nodes as the `TransactionManager` on worker nodes does not have correct information.

To implement organizer/compactor, we should have a pre-requirement for worker to begin a transaction.

Reply:
Right.  Worker doesn't call that right now. 

Worker does need transactions to track created_chunks, which we forgot in the beginning. added.